src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1575,19 +1577,20 @@ private TypeDeclarationSyntax ParseClassOrStructOrInterfaceOrExtensionDeclaratio
             var name = this.ParseIdentifierToken();
             var typeParameters = this.ParseTypeParameterList();
 
-            var paramList = CurrentToken.Kind == SyntaxKind.OpenParenToken && mainKeyword.Kind != SyntaxKind.ExtensionKeyword // PROTOTYPE
-                ? ParseParenthesizedParameterList() : null;
+            var paramList = CurrentToken.Kind == SyntaxKind.OpenParenToken
+                ? ParseParenthesizedParameterList()
```

Comment:
>ParseParenthesizedParameterList() [](http://example.com/codeflow?start=18&length=33)

Did we check binding code to make sure it doesn't try to create a symbol for the primary constructor for extension types?

Reply:
test added for this.